### ISSUE TITLE
autofocus

### ISSUE THREAD

**Issue by tlambert03** [pymmcore-plus/useq-schema#4]
as pointed out by @henrypinkard in https://github.com/micro-manager/pycro-manager/issues/266#issuecomment-809565527, the concept of autofocus should be somewhere

**Comment by ianhi** [comment]
Following up on my experiences from: https://github.com/pymmcore-plus/pymmcore-plus/issues/51#issuecomment-1397585287

I think adding stage descriptors to the base `ZPlan` would essentially accomplish this:

```python
    primary_focus_stage: str = None
    z_stack_stage: str = None
```

In the case of my example in the linked issue I would set `primary_focus_stage` to `PFS-Offset` and `z_stack_stage` to `Focus`.

The idea being:
1. If both are specified then primary is used to find the center, and z_stack to perform the z_stack.
2. If only one is given then use that for everything
3. If neither are given use whatever is returned by `core.getDefaultFocusDevice`

**Comment by tlambert03** [comment]
closed by #96
